pkg/defaults/deployment_defaults.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaults
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+
+	"knative.dev/serving/pkg/apis/serving"
+)
+
+const (
+	// GroupName is the group name for istio labels and annotations
+	GroupName = "service.istio.io"
+
+	// RevisionLabelKey is the label key attached to k8s resources to indicate
+	// which Revision triggered their creation.
+	RevisionLabelKey = GroupName + "/canonical-revision"
+
+	// ServiceLabelKey is the label key attached to a Route and Configuration indicating by
+	// which Service they are created.
+	ServiceLabelKey = GroupName + "/canonical-service"
+)
+
+// IstioDeployment is a wrapper around Deployment for setting Istio specific defaults
+type IstioDeployment struct {
+	appsv1.Deployment `json:",inline"`
+}
+
+// Verify that Deployment adheres to the appropriate interfaces.
+var (
+	// Check that Deployment can be defaulted.
+	_ apis.Defaultable = (*IstioDeployment)(nil)
+	_ apis.Validatable = (*IstioDeployment)(nil)
+)
+
+// SetDefaults implements apis.Defaultable
+func (r *IstioDeployment) SetDefaults(ctx context.Context) {
+	logger := logging.FromContext(ctx)
+	logger.Info("Calling SetDefaults")
```

Comment:
Do we need this log?

Reply:
Nope there's logging in the main controller. Removed.